torch/csrc/generic/StorageSharing.cpp

```diff
@@ -323,9 +323,6 @@ static PyObject * THPStorage_(newSharedCuda)(PyObject *_unused, PyObject *args)
 static PyObject * THPStorage_(weakRef)(THPStorage *self, PyObject *weak_ref_class) {
   HANDLE_TH_ERRORS
   THStorage* storage = self->cdata;
-  while (storage->flag & TH_STORAGE_VIEW) {
-    storage = storage->view;
-  }
```

Comment:
Why is this gone now? It's important. The view storages never use their allocator directly, so this should happen in the root one.

Reply:
Good point. I was looking for a way to make a weakref of a view storage (because view storages are being cached into "real cache"). Let me investigate to see if this is possible

Reply:
@apaszke I'm looking for a way to make a "weakref of a view storage" because I'd like to cache the view storage. Before this PR, the weakref function unwraps views and creates a weakref to the resulting storage. 

I'm thinking about one of the following two approaches:
- Attaching an `__offset` field to the `StorageRef`. The view storage can be recreated by taking `storage + __offset`, where `storage` is the result of unwrapping all views from the view storage.
- Changing the invariant that the "view storages never use their allocator directly" by making a ViewStorageAllocator or something along that line so that I can directly create a weakref to the view storage.

I'm leaning toward the first option because it doesn't require any changes to how storages work. What do you think?